zeebe/util/src/main/java/io/camunda/zeebe/util/collection/EnumTable.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.util.collection;
+
+import java.util.function.BiFunction;
+import java.util.function.IntFunction;
+
+/**
+ * A table implementation optimized for {@link Enum} keys. See {@link MapTable} for more on tables.
+ *
+ * <p>Enums allow us to define a finite keyspace, so we can efficiently use a single array to
```

Comment:
:wrench: We can also mention that we are using `row-major` ordering as well in the intro, so that it's quicker to double check if we are using row-major or column-major ordering. 
Great explanation anyway! :rocket: